src/service/ratelimit.go

```diff
@@ -240,6 +245,15 @@ func (this *service) ShouldRateLimit(
 	ctx context.Context,
 	request *pb.RateLimitRequest) (finalResponse *pb.RateLimitResponse, finalError error) {
 
+	// Generate trace
+	_, span := tracer.Start(ctx, "ShouldRateLimit Execution",
```

Comment:
What is the cost of all of these pan creations if tracing is disabled?

Reply:
When tracing is diabled, no span processor or exporter is added to the trace provider, hence tracer.Start and tracer.End are basically doing noop.